src/com/amazon/ion/impl/bin/IntList.java

```diff
@@ -0,0 +1,105 @@
+package com.amazon.ion.impl.bin;
+
+/**
+ * A list of integer values that grows as necessary.
+ *
+ * Unlike {@link java.util.List}, IntList does not require each int to be boxed. This makes it helpful in use cases
+ * where storing {@link Integer} leads to excessive time spent in garbage collection.
+ */
+public class IntList {
```

Comment:
We'll want to keep an eye on [JEP 401](https://openjdk.java.net/jeps/401) relevant to this, yeah?

Reply:
Secondly, this seems like it would be a good place to use HPPC, e.g. [IntArrayList](http://carrotsearch.github.io/hppc/releases/0.7.0/api/com/carrotsearch/hppc/IntArrayList.html). 

I know that opens the eternal question of when to implement and when to depend, but I do want to intentionally raise the question. Do we have enough primitive collection handling that something HPPC is worth the trade-off? What about e.g. stormpot for object pooling?

No action needed this PR.

Reply:
> We'll want to keep an eye on JEP 401 relevant to this, yeah?

Yep! Although we'd need to target a more modern JDK to benefit from it. 😞

> Secondly, this seems like it would be a good place to use HPPC, e.g. IntArrayList.
> I know that opens the eternal question of when to implement and when to depend, but I do want to intentionally raise the question. Do we have enough primitive collection handling that something HPPC is worth the trade-off? What about e.g. stormpot for object pooling?

AFAIK this is our only homebrew primitive collection. I'm all for deferring to a battle-tested dependency once we reach some sort of critical mass, but happily this time we only needed a very simple one-off data structure.